vendor/github.com/coreos/etcd/auth/store.go

```diff
@@ -441,7 +441,7 @@ func (as *authStore) UserGrantRole(r *pb.AuthUserGrantRoleRequest) (*pb.AuthUser
 	}
 
 	idx := sort.SearchStrings(user.Roles, r.Role)
-	if idx < len(user.Roles) && strings.Compare(user.Roles[idx], r.Role) == 0 {
+	if idx < len(user.Roles) && user.Roles[idx] == r.Role {
```

Comment:
You can't change anything directly in vendored dirs, you need to submit that directly to http://github.com/coreos/etcd, instead.